src/libraries/System.Private.CoreLib/src/System/WeakReference.cs

```diff
@@ -46,5 +47,95 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
         // or just until they're finalized (false).
         //
         public virtual bool TrackResurrection => IsTrackResurrection();
+
+#if !CORECLR
+        // Most methods using the handle should use GC.KeepAlive(this) to avoid potential handle recycling
+        // attacks (i.e. if the WeakReference instance is finalized away underneath you when you're still
+        // handling a cached value of the handle then the handle could be freed and reused).
+
+        // the handle field is effectively readonly until the object is finalized.
+        // the lowermost bit is used to indicate whether the handle is tracking resurrection
+        private IntPtr m_handleAndKind;
```

Comment:
```suggestion
        private IntPtr _handleAndKind;
```
Nit: to follow the libraries naming conventions

Reply:
Switched to `_handleAndKind` style.